src/backend/joanie/tests/core/test_api_contract.py

```diff
@@ -621,3 +629,447 @@ def test_api_contract_download_authenticated_should_fail_if_contract_is_not_sign
+
+    def test_api_contract_generate_zip_archive_authenticated_post_with_no_signed_contracts(
+        self,
+    ):
+        """
+        Authenticated user should be able to use POST method on the viewset to generate ZIP
+        archive when parsing an existing Organization UUID where the user has the rights to access,
+        but it won't generate a ZIP archive because there are no signed contracts.
+        """
+        user = factories.UserFactory()
+        organization = factories.OrganizationFactory()
+        factories.UserOrganizationAccessFactory(organization=organization, user=user)
+        token = self.get_user_token(user.username)
+
+        response = self.client.post(
+            "/api/v1.0/contracts/zip_archive/",
+            data={"organization_id": organization.id},
+            HTTP_AUTHORIZATION=f"Bearer {token}",
+        )
+
+        self.assertEqual(response.status_code, 400)
+
+        self.assertEqual(response.json(), ["No zip to generate"])
+
+    # pylint: disable=too-many-locals
+    def test_api_contract_generate_zip_archive_authenticated_post_method_allowed(self):
+        """
+        Authenticated user should be able to use POST method on the viewset to generate ZIP
+        archive when parsing an existing Organization UUID where the user has the rights to access,
+        and it will generate a ZIP archive with the signed contracts.
+        """
+        storage = storages["contracts"]
+        requesting_user = factories.UserFactory()
+        organization = factories.OrganizationFactory()
+        factories.UserOrganizationAccessFactory(
+            organization=organization, user=requesting_user
+        )
+        relation = factories.CourseProductRelationFactory(organizations=[organization])
+        learners = factories.UserFactory.create_batch(3)
+        signature_reference_choices = [
+            "wfl_fake_dummy_1",
+        ]
+        for index, reference in enumerate(signature_reference_choices):
+            user = learners[index]
+            order = factories.OrderFactory(
+                owner=user,
+                product=relation.product,
+                course=relation.course,
+                state=enums.ORDER_STATE_VALIDATED,
+            )
+            context = contract_definition.generate_document_context(
+                order.product.contract_definition, user, order
+            )
+            factories.ContractFactory(
+                order=order,
+                signature_backend_reference=reference,
+                definition_checksum="1234",
+                context=context,
+                signed_on=timezone.now(),
+            )
+        expected_endpoint_polling = "/api/v1.0/contracts/zip_archive/"
+        token = self.get_user_token(requesting_user.username)
+
+        response = self.client.post(
+            "/api/v1.0/contracts/zip_archive/",
+            data={"organization_id": organization.id},
+            HTTP_AUTHORIZATION=f"Bearer {token}",
+        )
```

Comment:
Just a thought,  shouldn't we use something like GET `/api/v1.0/organizations/<orga_id>/contracts/zip-archive` ?

Reply:
to create a zip, we should use a `POST`. And this endpoint can be used also in other context than an organization one.